File: src/easy/1-two-sum/python/main.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if (len(nums) == 2): return [0, 1]  # noqa: E701
        number_collection = {}
        for i, num in enumerate(nums):
            if (target - num in number_collection):
                return [number_collection[target - num], i]
            number_collection[num] = i

File: src/easy/1-two-sum/typescript/main.ts
/* eslint-disable curly */
/* eslint-disable unicorn/no-for-loop */
function twoSum(nums: number[], target: number): number[] {
	if (nums.length === 2) return [0, 1];

	const numberIndices = new Map<number, number>();
	for (let i = 0; i < nums.length; i++) {
		const currentNumber = nums[i];
		const complement = target - currentNumber;
		if (numberIndices.has(complement)) return [numberIndices.get(complement)!, i];

		numberIndices.set(currentNumber, i);
	}

	return [];
}
